### PR TITLE
tool: expire is a tool to scan the whole database to expire keys

### PR DESCRIPTION
**Pull request by shafreeck**
When the expire feature has a bug, it may cause dirty keys that expected to be expired. This is a tool to scan the whole database, reading its meta and clean it up if it has expired.

### CODE REVIEW

File: tools/expire/main.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+func cleanUp(txn kv.Transaction, mkey kv.Key, database *db.DB,
+	obj *db.Object) error {
+	switch obj.Type {
+	case db.ObjectString:
+		return txn.Delete(mkey)
+	case db.ObjectZSet:
+		dkey := db.DataKey(database, obj.ID)
+		skey := db.ZSetScorePrefix(database, obj.ID)
+		txn.Delete(dkey) // ignore its error for simplicity
+		return txn.Delete(skey)
```
Comment (nioshield) [inline tools/expire/main.go]: Delete prefix key when there are multiple members?

File: tools/expire/main.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	return nil
+}
+
+func doExpire(s kv.Storage, database *db.DB, start kv.Key, limit int) (kv.Key, error) {
+	txn, err := s.Begin()
+	if err != nil {
+		return nil, err
+	}
+	iter, err := txn.Iter(start, nil)
```
Comment (nioshield) [inline tools/expire/main.go]: defined end prefix key
 kv.Key(start).PrefixNext()

File: tools/expire/main.go (hunk 1 above)
Comment (shafreeck) [reply tools/expire/main.go]: Yes. I made a mistake.

File: tools/expire/main.go (hunk 2 above)
Comment (shafreeck) [reply tools/expire/main.go]: I want to scan the whole databases, so I did not want a prefix-next.

**Pull request merged**